pylint/checkers/format.py

```diff
@@ -430,9 +506,19 @@ def process_tokens(self, tokens: list[tokenize.TokenInfo]) -> None:
                 if check_equal:
                     check_equal = False
                     self.check_indent_level(line, indents[-1], line_num)
-
-            if tok_type == tokenize.NUMBER and string.endswith("l"):
-                self.add_message("lowercase-l-suffix", line=line_num)
+            if tok_type == tokenize.NUMBER:
+                if (
+                    self.linter.is_message_enabled("bad-float-notation")
+                    and
+                    # You don't deserve a linter if you mix non-decimal notation and
+                    # exponential or underscore,
+                    "x" not in string  # not a hexadecimal
+                    and "o" not in string  # not an octal
+                    and "j" not in string  # not a complex
```

Comment:
PEP 515 is talking about separating word in hexadecimal, might want to do something about it... later.